### ISSUE TITLE
NCCL TIMEOUT When i using axolotl full fine tuning mixtral7B x 8

### ISSUE THREAD

**Issue by dumpmemory** [axolotl-ai-cloud/axolotl#956]
### Please check that this issue hasn't been reported before.

- [X] I searched previous [Bug Reports](https://github.com/OpenAccess-AI-Collective/axolotl/labels/bug) didn't find any similar reports.

### Expected Behavior

no timeout

### Current behaviour

faced with nccl timeout during backward after 1:30 hours training

### Steps to reproduce

System Info
transformers version: 4.36.0
Platform: Linux-5.4.119-19.0009.28-x86_64-with-glibc2.35
Python version: 3.10.6
Huggingface_hub version: 0.19.4
Safetensors version: 0.4.1
Accelerate version: 0.23.0
Accelerate config: not found
PyTorch version (GPU?): 2.1.0a0+b5021ba (True)
Tensorflow version (GPU?): not installed (NA)
Flax version (CPU?/GPU?/TPU?): not installed (NA)
Jax version: not installed
JaxLib version: not installed
Using GPU in script?:
Using distributed or parallel set-up in script?: Yes

just run the example with my custom dataset

### Config yaml

using deepspeed zero3

### Which Operating Systems are you using?

- [X] Linux
- [ ] macOS
- [ ] Windows

### Python Version

3.10

### axolotl branch-commit

main

### Acknowledgements

- [X] My issue title is concise, descriptive, and in title casing.
- [X] I have searched the existing issues to make sure this bug has not been reported yet.
- [X] I am using the latest version of axolotl.
- [X] I have provided enough information for the maintainers to reproduce and diagnose the issue.

**Comment by dumpmemory** [comment]
after update nccl to [2.19.3](https://docs.nvidia.com/deeplearning/nccl/release-notes/rel_2-19-3.html#rel_2-19-3)  with nvidia 23-10, it was running 7 hours !
